test/test_cuda.py

```diff
@@ -5265,6 +5264,89 @@ def foo():
 
         self.checkFunction(foo, [])
 
+    def test_additional_free_following_checkpoint(self):
+
+        # put in closure to force deallocations
+        def func():
+            def foo():
+                return int8_cuda(MIN_BLOCK_SIZE),
+
+            def foo2():
+                return int8_cuda(MIN_BLOCK_SIZE),
+
+            graph, outputs = cudagraphify(foo, [])
+            pool_id = graph.pool()
+
+            segments_before_checkpoint = get_cudagraph_segments(pool_id)
+
+            state = torch._C._cuda_getCheckpointState(outputs[0].device.index, pool_id)
+
+            graph2, outputs2 = cudagraphify(foo2, [], pool=graph.pool())
+
+
+            torch._C._cuda_setCheckpointPoolState(outputs[0].device.index, state, outputs2)
+
+            del outputs2
+
+            self.checkCheckpointedState(segments_before_checkpoint, get_cudagraph_segments(pool_id))
+
+        func()
+
+        torch.cuda.synchronize()
+        gc.collect()
+        torch.cuda.empty_cache()
+
+    # TODO: re-enable
```

Comment:
You gonna reenable?

Reply:
There is another comment thread that I commented on this about. To enable this we would have to league Storages as a concept to the allocator, so i'm not sure. Explanation here: https://github.com/pytorch/pytorch/pull/94943#discussion_r1120749709.

Reply:
ok, just breadcrumb it in the comment here, so we can find it later